### PR TITLE
9.0 port account bank statement import ofx

### PR DESCRIPTION
**Pull request by legalsylvain**

### CODE REVIEW

**Comment by alexis-via** [comment]
Could you get the changes of this v8 PR https://github.com/OCA/bank-statement-import/pull/62

**Comment by gurneyalex** [comment]
#44 also does a port of this module

**Comment by legalsylvain** [comment]
@gurneyalex : yes my PR include https://github.com/OCA/bank-statement-import/pull/44 changes.

**Comment by legalsylvain** [comment]
@alexis-via : I just rebased my work. #62 should be included now.

**Comment by pedrobaeza** [comment]
Please check CIs status

**Comment by ericzen** [comment]
So what's the OCA's position ? ofx import is valide for odoo 9 ? And in case of yes, how can i correct the error describes in #74 ?
For information i use 9.0_port_account_bank_statement_import_ofx branch.

**Comment by pedrobaeza** [comment]
So you were referring to this PR? Any comment for something in progress should be reported here. Anyways, you don't describe any error. You should provide the error in the Odoo log and ideally the file you're trying to import to see the problem.

**Comment by legalsylvain** [comment]
Hi @ericzen.

Thanks for testing. we are currently using this module without problem. So any log or detailled description of your problem are welcome ! 

regards.

**Comment by ericzen** [comment]
unforgattable, there is no error in odoo's log (odoo-server.log). I've only this message box. That's why i give it to you.
I've test with demo.ofx files and same error.
Perhaps due to my install method.
Can you give me your method to install your addons ?
what i've done 
install python ofxparse
In settings/apps install account_bank_statement_import, account_bank_statement_import_multiline, account_bank_statement_import_ofx

**Comment by ericzen** [comment]
Something can help me if you know what is Bank Reference. Indeed, perhaps my journal setting or Bank account setting is wrong.

**Comment by ericzen** [comment]
OK It's work
Create a new folder and install account_bank_statement_import, account_bank_statement_import_ofx
work's fine now
Thanks

**Review by alexis-via — APPROVED** [review]
:+1:  when travis will be green. If we port to v10, we should start from here.

**Comment by pedrobaeza** [comment]
@legalsylvain please check Travis status

**Comment by alexis-via** [comment]
According to a very rapid test, this version of the ofx module also works on v10 without any modification (I tested it with my module account_bank_statement_import_usability available here for v10 https://github.com/akretion/odoo-usability/tree/10.0/account_bank_statement_import_usability that matches bank account when the OFX files has the national account number and Odoo has the IBAN)

**Comment by NL66278** [comment]
The tests fail because of this issue: https://github.com/OCA/bank-statement-import/issues/76 Must have something to do with the version of ofxparse I think. We might need to pin a certain version of this, but I do not know which would be the right one.

**Comment by alexis-via** [comment]
I confirm this version works fine on v10 (tested with 2 different French banks, BNP and Banque populaires). I use ofxparse 0.15, which is the latest version cf https://pypi.python.org/pypi/ofxparse

**Comment by pedrobaeza** [comment]
But please fix Travis

**Comment by pedrobaeza** [comment]
It continues the same...

**Comment by alexis-via** [comment]
It is very strange, because with the latest version of ofxparse from pypi, I don't have this pb.

File: account_bank_statement_import_ofx/demo/demo_data.xml [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
-    </data>
 
-</openerp>
+    <record id="ofx_bank_journal" model="account.journal">
+        <field name="name">Bank Journal - (test ofx)</field>
+        <field name="bank_acc_number">123456</field>
+        <field name="code">TBNKOFX</field>
+        <field name="type">bank</field>
+        <field name="sequence_id" ref="ofx_sequence"/>
+        <field name="user_id" ref="base.user_root"/>
```
Comment (hbrunn) [inline account_bank_statement_import_ofx/demo/demo_data.xml:19]: this field doesn't exist any more

**Review by hbrunn — CHANGES_REQUESTED** [review]
unfortunately, I can't help with the travis problem. A quick glance at https://github.com/jseutter/ofxparse/blob/master/ofxparse/ofxparse.py seems like the balance attribute is only set under certain conditions, any chance the test ofx file for the test simply doesn't contain whatever https://github.com/jseutter/ofxparse/blob/master/ofxparse/ofxparse.py#L910 wants?

**Comment by alexis-via** [comment]
Now that ofxparse 0.15 has been release on pypi (cf https://pypi.python.org/pypi/ofxparse/0.15 released on June 30 2016, the date of your PR !), you can remove the file account_bank_statement_import_ofx/models/ofx.py
I confirm ofxparse 0.15 includes the fix
https://github.com/jseutter/ofxparse/commit/283f89c3246ed3fedccc3ef5c96078b7d5b94579

**Comment by alexis-via** [comment]
BTW, I pushed a small PR to sylvain's branch: https://github.com/legalsylvain/bank-statement-import/pull/1 to update this PR (not to remove the OFX parser, but for other small things).

**Comment by alexis-via** [comment]
I changed many more things in my PR of the port of the OFX module for v10. Maybe it will be a good backport them here once they are finished and stable.

**Comment by legalsylvain** [comment]
Hi all, I'll merge PR of @alexis-via . I have no time to make the requested changes immediately. (removing obsolete ofx patch, ...) I'll review and accept new PR against this PR, or I'll do the changes in a while.
kind regards.

File: account_bank_statement_import_ofx/models/account_bank_statement_import.py [hunk 2]
```diff
@@ -38,15 +35,23 @@ def _parse_file(self, data_file):
...
+                    continue
                 # Since ofxparse doesn't provide account numbers, we'll have
                 # to find res.partner and res.partner.bank here
                 # (normal behavious is to provide 'account_number', which the
                 # generic module uses to find partner/bank)
                 bank_account_id = partner_id = False
                 banks = self.env['res.partner.bank'].search(
-                    [('owner_name', '=', transaction.payee)], limit=1)
+                    [('bank_name', '=', transaction.payee)], limit=1)
                 if banks:
```
Comment (alexis-via) [inline account_bank_statement_import_ofx/models/account_bank_statement_import.py:55]: This is wrong ; we should  keep the search on owner_name.

**Comment by alexis-via** [comment]
@legalsylvain I think it will be easier to backport my v10 version to v9 when the v10 version will be finalized, as we continue to discover/fix issues in this PR during the review of the v10 PR https://github.com/OCA/bank-statement-import/pull/80. So let's continue the discussion on the v10 PR.

**Comment by legalsylvain** [comment]
@alexis-via : Agree. Bette workflow. let keep it open this PR for the time being, to avoid other people working on porting this module in V9.0. In the meantime, let's go to review to V10.0 PR, and we'll backport your work.

Thanks for your work.

**Comment by flotho** [comment]
Hi everyone,
As I can see the v10 migration has been merged.
Any chance to backport the job in this PR?
BTW, I tried this PR with the ofxparse from github master and still have the message :
```
    'balance_start': ofx.account.statement.balance,
AttributeError: 'Statement' object has no attribute 'balance'
```
Does the #80 solved this kind of issue?
Regards

**Comment by alexis-via** [comment]
Yes, in #80 Travis is green, we don't have the error "balance_start': ofx.account.statement.balance,
AttributeError: 'Statement' object has no attribute 'balance'" any more. But we fixed this message by taking the last version of ofxparse from github... so it's strange that you still have it. And, in my case, I never had this message when using account_bank_statement_import_ofx in real life with real OFX files from several French banks using ofxparse 0.14 or 0.15... so for me it was a Travis-only issue.

**Comment by flotho** [comment]
Hi @alexis-via , I personnaly tested the importation of 2 OFX from the same french bank group Credit Agricole. I have 2 files depending on different "federation". One of them is correctly imported the other one throw the famous "Statement' object has no attribute 'balance'"" error.
I've tested the importation with both ofxparse 0.15 from pypi and 0.14 from the github .
Is there any additionnal part of code that has changed /not been backported from v10 that could explain this behaviour?
Regards

**Comment by HeO2SatDvr** [comment]
Hi Everyone.  I would like to contribute to this by testing with Canadian and US bank files but I don't know where to begin.  I have Odoo 9 community installed and functioning.  I have not updated the code for Odoo for about 9 months.  Do I need to have the latest Odoo code?  How and where to I install ofxparse?  How and where to I install the Odoo Addon developed here?

I am not a programmer.  I am a Management Consultant and I primarily work on Oracle systems.  I am exploring Odoo and thought testing might be a good way to contribute.

Thanks.

**Comment by pedrobaeza** [comment]
Hi, @HeO2SatDvr, you can try the module using the guide in https://odoo-community.org/page/review, but unfortunately the problem here is a technical one I'm afraid.

**Pull request closed without merging**